File: backup/Invest-Shop/Invest-Shop-Repository/src/main/java/com/illunex/invest/license/repository/ProductRepository.java
package com.illunex.invest.license.repository;

import com.illunex.invest.license.entity.Product;
import com.illunex.invest.license.enumable.ProductType;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface ProductRepository extends JpaRepository<Product, Long> {
    List<Product> findAllByProductType(ProductType productType);
}

File: microservices/shop/src/main/kotlin/com/illunex/invest/shop/persistence/repository/ProductRepository.kt
package com.illunex.invest.shop.persistence.repository

import com.illunex.invest.shop.persistence.entity.Product
import com.illunex.invest.shop.persistence.entity.enumable.ViewMode
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface ProductRepository: JpaRepository<Product, Long> {
    fun findByViewMode(viewMode: ViewMode): List<Product>
}